schema/deploy/mutations/generate_quarterly_reports.sql

```diff
@@ -0,0 +1,56 @@
+-- Deploy cif:mutations/generate_quarterly_reports to pg
+
+begin;
+
+create or replace function cif.generate_quarterly_reports(revision_id int, start_date timestamptz, end_date timestamptz)
+  returns setof cif.form_change as
+  $generate_quarterly_reports$
+declare
+  start_date_year int := extract(year from $2);
+  report_interval_start_date timestamptz;
+begin
+  if $2 >= $3 then
+      raise exception 'start_date must be before end_date';
+  end if;
+
+  -- choosing the closest quarter start date from a list of possible dates as a temporary table
+  select * from generate_series(make_date(start_date_year, 01, 05), make_date(start_date_year + 1, 01, 05), '3 month'::interval) as quarterly_report_due_date
+    where $2 <= quarterly_report_due_date limit 1 into report_interval_start_date;
```

Comment:
Very cool, didn't know you could select the first instance of match from a list like this